core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -422,6 +424,20 @@ private void configureRetryPolicy(SdkClientConfiguration.Builder config) {
     private void configureRetryStrategy(SdkClientConfiguration.Builder config) {
         RetryStrategy strategy = config.option(SdkClientOption.RETRY_STRATEGY);
         if (strategy != null) {
+            // Fix the retry strategy if no retry predicates were configured. Users using
+            // DefaultRetryStrategy do not have any conditions configured as that package
+            // does not know anything about the SDK or AWS retry conditions. This mimics the
+            // retry policies behavior that are configured behind the scenes and avoids the
+            // risk of customers inadvertently using a no-op retry strategy.
+            if (strategy.maxAttempts() > 1
```

Comment:
Why do we need `maxAttempts() > 1` again?

Reply:
Not really needed, but avoids changing the strategy for when the user meant to avoid retries.